google-cloud-error_reporting/docs/instrumentation.md

```diff
@@ -1,39 +1,150 @@
+Then the instrumentation library can also be configured through Rails configuration interface:
 ```ruby
-# Sharing authentication parameters
-config.google_cloud.project_id = "gcp-project-id"
-config.google_cloud.keyfile = "/path/to/gcp/secret.json"
-# Or more specificly for ErrorReporting
-config.google_cloud.error_reporting.project_id = "gcp-project-id"
-config.google_cloud.error_reporting.keyfile = "/path/to/gcp/sercret.json"
- 
-# Explicitly enable or disable ErrorReporting
-config.google_cloud.use_error_reporting = true
+# In config/environments/*.rb
  
-# Set Stackdriver Error Reporting service context
-config.google_cloud.error_reporting.service_name = "my-app-name"
-config.google_cloud.error_reporting.service_version = "my-app-version"
+Rails.application.configure do
+    # Sharing authentication parameters
+    config.google_cloud.project_id = "gcp-project-id"
+    config.google_cloud.keyfile = "/path/to/gcp/secret.json"
+    # Or more specificly for ErrorReporting
+    config.google_cloud.error_reporting.project_id = "gcp-project-id"
+    config.google_cloud.error_reporting.keyfile = "/path/to/gcp/sercret.json"
+     
+    # Explicitly enable or disable ErrorReporting
+    config.google_cloud.use_error_reporting = true
+     
+    # Set Stackdriver Error Reporting service context
+    config.google_cloud.error_reporting.service_name = "my-app-name"
+    config.google_cloud.error_reporting.service_version = "my-app-version"
+end
 ```
 
 Alternatively, check out [stackdriver](https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/stackdriver) gem, which enables this Railtie by default.
 
-## Rack Integration
+### Rack Integration
 
-Other Rack base framework can also directly leverage the built-in Middleware:
+Other Rack base framework can also directly leverage the Middleware directly:
 ```ruby
-require "google/cloud/error_reporting/v1beta1"
+require "google/cloud/error_reporting"
+ 
+Google::Cloud::ErrorReporting.configure do |config|
+  config.project_id      = "my-project"
+  config.keyfile         = "/path/to/keyfile.json"
+  config.service_name    = "my-app-name"
+  config.service_version = "my-app-version"
+end
  
 use Google::Cloud::ErrorReporting::Middleware
 ```
+
+## Report Captured Expcetions
+Captured Ruby exceptions can be directly reported to Stackdriver ErrorReporting
+by using `Google::Cloud::ErrorReporting.report`
+```ruby
+begin
+  fail "Boom!"
+rescue => exception
+  Google::Cloud::ErrorReporting.report exception
+end
+```
+
+The reported error event can also be customized
+```ruby
+begin
+  fail "Boom!"
+rescue => exception
+  Google::Cloud::ErrorReporting.report exception do |error_event|
+    # Directly modify the Google::Cloud::ErrorReporting::ErrorEvent object before submission
+    error_event.message     = "Custom error message"
+    error_event.user        = "johndoh@example.com"
+    error_event.http_status = 502
+  end
+end
+```
```

Comment:
Should be fixed in https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1454/commits/f7c436756604ca46d2e752c29162a94ea7b554ab